driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java

```diff
@@ -437,21 +437,12 @@ static boolean canRetryWrite(final ServerDescription serverDescription, final Co
         return true;
     }
 
-    static boolean isRetryableRead(final boolean retryReads, final ServerDescription serverDescription,
```

Comment:
In JAVA-4034, use of this method was removed with the addition of the `RetryState` class, but that caused us to lost the check for `sessionContext.hasActiveTransaction()`.  Since the `RetryState` class already has the check for whether `retryReads` is enabled, I chose to just remove this unused method and move that check into `canRetryRead` below.